### PR TITLE
Add Immutable.JS

### PR DESCRIPTION
**Pull request by JeremD**

### CODE REVIEW

**Comment by JeremD** [comment]
Hey ! I made this PR to add ImmutableJS (scoped to the app reducers), as it has been added to the theodo stack article :)
I would be happy to discuss my PR (first one I do on a generator)

File: generators/app/templates/client/src/components/Root/Root.container.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { connect } from 'react-redux';
+import { toJs } from '../../hocs/toJs';
```
Comment (cRicateau) [inline generators/app/templates/client/src/components/Root/Root.container.js]: hocs ?

File: generators/app/templates/client/src/hocs/toJs.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Iterable } from 'immutable';
+
+export const toJS = WrappedComponent => wrappedComponentProps => {
```
Comment (cRicateau) [inline generators/app/templates/client/src/hocs/toJs.js]: why do you need such a component ?

Comment (cRicateau) [reply generators/app/templates/client/src/hocs/toJs.js]: what is your point of view ? should we inject JS in component or use immutable in component ?

File: generators/app/templates/client/src/components/Root/Root.container.js (hunk 1 above)
Comment (JeremD) [reply generators/app/templates/client/src/components/Root/Root.container.js]: Higher order components -> I didn't know where to put this reusable bit of code, where would you put it ?

File: generators/app/templates/client/src/hocs/toJs.js (hunk 2 above)
Comment (JeremD) [reply generators/app/templates/client/src/hocs/toJs.js]: This component allows for an immutable state and immutable methods in the reducers without forcing immutable data-structures in the rest of the code. 

For me both could be interesting but this is the solution we describe in the stack article and answers the problem 'How do we enforce state immutability' but does not impact too much of the code base.

the HOC pattern for conversion is proposed by redux [here](https://redux.js.org/docs/recipes/UsingImmutableJS.html#use-a-higher-order-component-to-convert-your-smart-components-immutablejs-props-to-your-dumb-components-javascript-props)

File: generators/app/templates/client/src/components/Root/Root.container.js (hunk 1 above)
Comment (cRicateau) [reply generators/app/templates/client/src/components/Root/Root.container.js]: services/immutable ?

File: generators/app/templates/client/src/hocs/toJs.js (hunk 2 above)
Comment (cRicateau) [reply generators/app/templates/client/src/hocs/toJs.js]: OK i got it, could you add the link to the article in the file to explain the why of the code ?

File: generators/app/index.js [hunk 3]
```diff
@@ -136,7 +136,8 @@ class StackGenerator extends Generator {
         'redux':'3.7.2',
         'redux-saga': '0.15.6',
         'source-map-explorer': '1.4.0',
-        'whatwg-fetch': '2.0.3'
+        'whatwg-fetch': '2.0.3',
+        'immutable': '4.0.0'
```
Comment (cRicateau) [inline generators/app/index.js]: version 3.8.2 => 4.0.0 doesnt exist (4.0.0-rc... exists) => use latest from npm https://www.npmjs.com/package/immutable

**Pull request merged**